src/LercLib/Lerc.cpp

```diff
@@ -462,3 +474,47 @@ ErrCode Lerc::ConvertToDoubleTempl(const T* pDataIn, size_t nDataValues, double*
 
 // -------------------------------------------------------------------------- ;
 
+template<class T> static ErrCode Lerc::CheckForNaN(const T* arr, int nDim, int nCols, int nRows, const BitMask* pBitMask)
+{
+  if (!arr || nDim <= 0 || nCols <= 0 || nRows <= 0)
+    return ErrCode::WrongParam;
+
+  const bool isDbl = (typeid(*arr) == typeid(double));

```

Comment:
I guess there would be a more template'ish way of making sure this method is only defined for float and double, and a no-op for other types. Which would save the isFlt ternary tests and cast below